WordPressKit/WordPressKit/PostServiceRemoteREST.m

```diff
@@ -119,6 +119,44 @@ - (void)createPost:(RemotePost *)post
            }];
 }
 
+- (void)createPost:(RemotePost *)post
+         withMedia:(RemoteMedia *)media
+    requestEqueued:(void (^)(void))requestEqueued
+           success:(void (^)(RemotePost *))success
+           failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([post isKindOfClass:[RemotePost class]]);
+
+    NSString *type = media.mimeType;
+    NSString *filename = media.file;
+    NSString *path = [NSString stringWithFormat:@"sites/%@/posts/new", self.siteID];
+    NSString *requestUrl = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+
+    NSMutableDictionary *parameters = [NSMutableDictionary dictionaryWithDictionary:@{}];
+    parameters[@"content"] = post.content;
+    parameters[@"title"] = post.title;
+    parameters[@"status"] = post.status;
+    FilePart *filePart = [[FilePart alloc] initWithParameterName:@"media[]" url:media.localURL filename:filename mimeType:type];
+    [self.wordPressComRestApi multipartPOST:requestUrl
+                                 parameters:parameters
+                                  fileParts:@[filePart]
+                                    success:^(id  _Nonnull responseObject, NSHTTPURLResponse * _Nullable httpResponse) {
+        RemotePost *post = [self remotePostFromJSONDictionary:responseObject];
+        if (success) {
+            success(post);
+        }
+    } failure:^(NSError *error, NSHTTPURLResponse *httpResponse) {
+        if (failure) {
+            failure(error);
+        }
+    }];
+
+    if (requestEqueued) {
```

Comment:
I'm not sure if this is the right moment to invoke the callback if you check inside the WordPressComRestApi class when you call multipart before we enqueue the request we are serialising the fileParts to a file. This process can take some time, so if you return too early an close the extension the serialisation can get killed before the request is added to the background session.

Reply:
Nice catch, thank you. Soooo, I will have to bubble something up from within the `WordPressComRestApi` call post-serialization.

Reply:
I think so, in order to be sure the file is on the hands of the background session.